File: DataBindingLiveDataJava/app/src/main/java/com/websarva/wings/android/databindinglivedatajava/data/model/Rectangle.java
package com.websarva.wings.android.databindinglivedatajava.data.model;

public class Rectangle {
	private final int _height;
	private final int _width;

	public Rectangle(int height, int width) {
		_height = height;
		_width = width;
	}

	public int getArea() {
		return _height * _width;
	}

	public String getAreaStr() {
		return String.valueOf(getArea());
	}

	public int getHeight() {
		return _height;
	}
	public int getWidth() {
		return _width;
	}

	public String getHeightStr() {
		return String.valueOf(_height);
	}

	public String getWidthStr() {
		return String.valueOf(_width);
	}
}

File: DataBinding2WayKotlin/app/src/main/java/com/websarva/wings/android/databinding2waykotlin/data/model/Rectangle.kt
package com.websarva.wings.android.databinding2waykotlin.data.model

import androidx.databinding.BaseObservable
import androidx.databinding.Bindable
import com.websarva.wings.android.databinding2waykotlin.BR

class Rectangle: BaseObservable() {
	var height: Int = 0
		set(value) {
			field = value
			notifyPropertyChanged(BR.heightStr)
			notifyPropertyChanged(BR.areaStr)
		}
	var width: Int = 0
		set(value) {
			field = value
			notifyPropertyChanged(BR.widthStr)
			notifyPropertyChanged(BR.areaStr)
		}

	fun getArea(): Int {
		return height * width
	}
	@Bindable
	fun getAreaStr(): String {
		return getArea().toString()
	}

	@Bindable
	fun getHeightStr(): String {
		return height.toString()
	}

	@Bindable
	fun getWidthStr(): String {
		return  width.toString()
	}

	@Bindable
	fun setHeightStr(value: String) {
		if(value != "") {
			val valueInt = value.toInt()
			height = valueInt
		}
	}

	@Bindable
	fun setWidthStr(value: String) {
		if(value != "") {
			val valueInt = value.toInt()
			width = valueInt
		}
	}
}
